### PR TITLE
Cortx 29598 f 45 i tests

### PR DESCRIPTION
**Pull request by kachhwahadivya**
# Problem Statement
- Problem statement

# Design
-  For Bug, Describe the fix here.
-  For Feature, Post the link for design

# Coding
   Checklist for Author
-  [ ] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
-  [ ] New/Affected tests are executed on Latest Build
-  [ ] Attach test execution logs
-  [ ] Collection tested and no collection error introduced (`pytest --local True --collect-only`)

# Review Checklist 
  Checklist for Author
- [ ] JIRA number/GitHub Issue added to PR
- [ ] PR is self reviewed
- [ ] Jira and state/status is updated and JIRA is updated with PR link
- [ ] Check if the description is clear and explained

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
-  [ ] If change in any common function, make sure to update all calls and execute all affected tests.

# Documentation
  Checklist for Author
-  [ ] Changes done to ReadMe / WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

**Comment by kachhwahadivya** [comment]
Passed logs attached to [[CORTX-29598] TA: F-45I- Automate 6 tests - JIRA NSS (seagate.com)](https://jts.seagate.com/browse/CORTX-29598)

File: tests/csm/rest/test_iam_users.py [hunk 1]
```diff
@@ -2127,6 +2126,190 @@ def test_39217(self):
...
+        """
+        test_case_name = cortxlogging.get_frame()
+        self.log.info("##### Test started -  %s #####", test_case_name)
+        self.log.info("Step-1: Creating IAM user")
+        payload = self.csm_obj.iam_user_payload_rgw("valid")
+        resp = self.csm_obj.create_iam_user_rgw(payload)
+        self.log.info("Verify Response : %s", resp)
+        assert resp.status_code == HTTPStatus.CREATED, "IAM user creation failed"
+        uid = resp.json()["user_id"]
+        self.created_iam_users.add(uid)
```
Comment (swapnil-khandare) [inline tests/csm/rest/test_iam_users.py:2146]: use tenant with uid while adding it to self.created_iam_users,   resp.json()["tenant"] + "$" + uid

Comment (swapnil-khandare) [reply tests/csm/rest/test_iam_users.py:2146]: Same is applicable for other tests

File: tests/csm/rest/test_iam_users.py [hunk 2]
```diff
@@ -2127,6 +2126,190 @@ def test_39217(self):
...
+
+
+    @pytest.mark.csmrest
+    @pytest.mark.lc
+    @pytest.mark.cluster_user_ops
+    @pytest.mark.parallel
+    @pytest.mark.tags('TEST-38918')
+    def test_38918(self):
+        """
+        Attempt to modify Access key, secret key while IO operations are ongoing with old key
```
Comment (swapnil-khandare) [inline tests/csm/rest/test_iam_users.py:2206]: we may need to change this test summary as we are not doing modification while ios are ongoing.

File: tests/csm/rest/test_iam_users.py (hunk 1 above)
Comment (kachhwahadivya) [reply tests/csm/rest/test_iam_users.py:2146]: Done

File: tests/csm/rest/test_iam_users.py (hunk 2 above)
Comment (kachhwahadivya) [reply tests/csm/rest/test_iam_users.py]: Done

File: tests/csm/rest/test_iam_users.py [hunk 3]
```diff
@@ -2127,6 +2126,190 @@ def test_39217(self):
...
+        payload = self.csm_obj.iam_user_payload_rgw("valid")
+        resp = self.csm_obj.create_iam_user_rgw(payload)
+        self.log.info("Verify Response : %s", resp)
+        assert resp.status_code == HTTPStatus.CREATED, "IAM user creation failed"
+        uid = resp.json()["tenant"] + "$" + resp.json()["user_id"]
+        self.created_iam_users.add(uid)
+
+        self.log.info("Step-2: Creating IAM user with same name as tenant")
+        payload = self.csm_obj.iam_user_payload_rgw("valid")
+        payload.update({"tenant": uid})
```
Comment (swapnil-khandare) [inline tests/csm/rest/test_iam_users.py:2150]: as uid is modified above now, will that work for issue which you saw OR you need user_id only here without resp.json()["tenant"] + "$"

File: tests/csm/rest/test_iam_users.py (hunk 1 above)
Comment (swapnil-khandare) [reply tests/csm/rest/test_iam_users.py:2146]: can you do this for test_38969 as well

File: tests/csm/rest/test_iam_users.py (hunk 3 above)
Comment (kachhwahadivya) [reply tests/csm/rest/test_iam_users.py]: Thanks for the catch. I was giving problem. Fixed now.

File: tests/csm/rest/test_iam_users.py (hunk 1 above)
Comment (kachhwahadivya) [reply tests/csm/rest/test_iam_users.py:2146]: done

**Review by swapnil-khandare — APPROVED** [review]

File: tests/csm/rest/test_iam_users.py [hunk 4]
```diff
@@ -2127,6 +2126,190 @@ def test_39217(self):
...
+        size = SystemRandom().randrange(10, 100)
+        file_path_upload = os.path.join(TEST_DATA_FOLDER, test_file)
+        if os.path.exists(file_path_upload):
+            os.remove(file_path_upload)
+        system_utils.create_file(file_path_upload, size)
+        resp = s3_obj.put_object(bucket_name=bucket, object_name=test_file,
+                    file_path=file_path_upload)
+
+        payload = {"access_key": usr["keys"][0]["access_key"],
+                    "secret_key":"sdss"}
```
Comment (aditipande96) [inline tests/csm/rest/test_iam_users.py:2235]: Can this secret key be fetched from config instead of hardcoding it?

Comment (kachhwahadivya) [reply tests/csm/rest/test_iam_users.py]: Replaced it with random key generated.

**Comment by RajeshPratapSInghChouhan** [comment]
please check DCO error, if possible can force push with 1 commit only ?

**Review by RajeshPratapSInghChouhan — APPROVED** [review]

**Review by aditipande96 — APPROVED** [review]

**Review by sonalk0209 — APPROVED** [review]
@kachhwahadivya  please add collections output
Approving for merge.

**Pull request merged**